src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/use_row_header_components.tsx

```diff
@@ -0,0 +1,523 @@
+  groupId: string;
+  groupValue: string;
+}
+
+interface RowClickActionContext {
+  dataView: DataView;
+  editorQuery: AggregateQuery;
+  statsFieldSummary: StatsCommandSummary['grouping'] | undefined;
+  esqlVariables: ESQLControlVariable[] | undefined;
+  rowContext: RowContext;
+  services: UnifiedDataTableProps['services'];
+  closeActionMenu: () => void;
+  globalState: TabStateGlobalState;
+  openInNewTab: (...args: Parameters<typeof internalStateActions.openInNewTab>) => void;
+  updateESQLQuery: DiscoverStateContainer['actions']['updateESQLQuery'];
+}
+
+/**
+ * Defines the context menu actions for the row header.
+ */
+const contextRowActions: Array<
+  EuiContextMenuPanelItemDescriptorEntry & {
+    /**
+     * The stats function type to render the action for. If not provided, the action will be rendered for all stats function types.
+     */
+    renderFor?: SupportedStatsFunction;
+    /**
+     * Allows an action to be enabled when the field is filterable.
+     */
+    enabledWhenFilterable?: boolean;
+  }
+> = [
+  {
+    name: i18n.translate('discover.esql_data_cascade.row.action.copy_to_clipboard', {
+      defaultMessage: 'Copy to clipboard',
+    }),
+    icon: 'copy',
+    'data-test-subj': 'dscCascadeRowContextActionCopyToClipboard',
+    onClick(this: RowClickActionContext) {
+      copyToClipboard(this.rowContext.groupValue as string);
+      return this.closeActionMenu();
+    },
+  },
+  {
+    enabledWhenFilterable: true,
+    name: i18n.translate('discover.esql_data_cascade.row.action.filter_in', {
+      defaultMessage: 'Filter in',
+    }),
+    icon: 'plusInCircle',
+    'data-test-subj': 'dscCascadeRowContextActionFilterIn',
+    onClick(this: RowClickActionContext) {
+      const updatedQuery = appendFilteringWhereClauseForCascadeLayout(
+        this.editorQuery.esql,
+        this.esqlVariables,
+        this.dataView,
+        this.rowContext.groupId,
+        this.rowContext.groupValue as string,
+        '+'
+      );
+
+      if (!updatedQuery) {
+        return;
+      }
+
+      this.updateESQLQuery(updatedQuery);
+
+      return this.closeActionMenu();
+    },
+  },
+  {
+    enabledWhenFilterable: true,
+    name: i18n.translate('discover.esql_data_cascade.row.action.filter_out', {
+      defaultMessage: 'Filter out',
+    }),
+    icon: 'minusInCircle',
+    'data-test-subj': 'dscCascadeRowContextActionFilterOut',
+    onClick(this: RowClickActionContext) {
+      const updatedQuery = appendFilteringWhereClauseForCascadeLayout(
+        this.editorQuery.esql,
+        this.esqlVariables,
+        this.dataView,
+        this.rowContext.groupId,
+        this.rowContext.groupValue as string,
+        '-'
+      );
+
+      if (!updatedQuery) {
+        return;
+      }
+
+      this.updateESQLQuery(updatedQuery);
+      return this.closeActionMenu();
+    },
+  },
+  {
+    name: i18n.translate('discover.esql_data_cascade.row.action.view_docs', {
+      defaultMessage: 'Open in new discover tab',
```

Comment:
```suggestion
      defaultMessage: 'Open in new tab',
```
"Discover" should be capitalized here, but IMO we should drop it entirely.